# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/extensions/webrequest/ChannelWrapper.cpp
# Commit: f2df73fda468
# Full Hash: f2df73fda46859150605f33bcdaeb49ab7037720
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-10-19 21:40:34
# Regressor Bug: 1588899
# File Overlap Count: 1
# Description:
#   Bug 1588899 - P1. Move classification flags related method to nsIClassifiedChannel. r=Ehsan,baku
#   
#   This is where it should have been in the first place. Those attributes belong there.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49577
# ==============================================================================

diff -r b98968eb71d3 -r f2df73fda468 toolkit/components/extensions/webrequest/ChannelWrapper.cpp
--- a/toolkit/components/extensions/webrequest/ChannelWrapper.cpp	Thu Oct 17 15:24:31 2019 +0000
+++ b/toolkit/components/extensions/webrequest/ChannelWrapper.cpp	Sat Oct 19 04:30:24 2019 +0000
@@ -24,6 +24,7 @@
 #include "mozilla/dom/Event.h"
 #include "mozilla/dom/BrowserHost.h"
 #include "nsIContentPolicy.h"
+#include "nsIClassifiedChannel.h"
 #include "nsIHttpChannelInternal.h"
 #include "nsIHttpHeaderVisitor.h"
 #include "nsIInterfaceRequestor.h"
@@ -48,7 +49,7 @@
 #define CHANNELWRAPPER_PROP_KEY \
   NS_LITERAL_STRING("ChannelWrapper::CachedInstance")
 
-using CF = nsIHttpChannel::ClassificationFlags;
+using CF = nsIClassifiedChannel::ClassificationFlags;
 using MUC = MozUrlClassificationFlags;
 
 struct ClassificationStruct {
@@ -906,14 +907,20 @@
 void ChannelWrapper::GetUrlClassification(
     dom::Nullable<dom::MozUrlClassification>& aRetVal, ErrorResult& aRv) const {
   MozUrlClassification classification;
-  if (nsCOMPtr<nsIHttpChannel> chan = MaybeHttpChannel()) {
+  nsCOMPtr<nsIHttpChannel> chan = MaybeHttpChannel();
+  nsCOMPtr<nsIClassifiedChannel> classified = do_QueryInterface(chan);
+  MOZ_DIAGNOSTIC_ASSERT(
+      classified,
+      "Must be an object inheriting from both nsIHttpChannel and "
+      "nsIClassifiedChannel");
+  if (classified) {
     uint32_t classificationFlags;
-    chan->GetFirstPartyClassificationFlags(&classificationFlags);
+    classified->GetFirstPartyClassificationFlags(&classificationFlags);
     FillClassification(classification.mFirstParty, classificationFlags, aRv);
     if (aRv.Failed()) {
       return;
     }
-    chan->GetThirdPartyClassificationFlags(&classificationFlags);
+    classified->GetThirdPartyClassificationFlags(&classificationFlags);
     FillClassification(classification.mThirdParty, classificationFlags, aRv);
   }
   aRetVal.SetValue(std::move(classification));